.github/workflows/check.yml

```diff
@@ -140,3 +140,24 @@ jobs:
       - name: cargo-semver-checks
         if: matrix.rust-toolchain == 'stable'
         run: cargo semver-checks
+
+  check-freebsd:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+      - name: Get "Install Rust" action from neqo
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          repository: mozilla/neqo
+          sparse-checkout: |
+            .github/actions/rust
+          path: neqo
+
+      - name: Install Rust
+        uses: ./neqo/.github/actions/rust
+
+      - run: rustup target add x86_64-unknown-freebsd
```

Comment:
Can't we do this step by extending the Rust action to support the `targets` input? https://github.com/dtolnay/rust-toolchain?tab=readme-ov-file#inputs

Reply:
> Can't we do this step by extending our Rust action to support the `targets` input? https://github.com/dtolnay/rust-toolchain?tab=readme-ov-file#inputs
> 
> (Should we move our Rust action to its own repo, while we're at it?)

Good idea. I will do that.

> (Should we move our Rust action to its own repo, while we're at it?)

I would actually suggest moving into the opposite direction, i.e. only move things into a new repo when technically necessary, defaulting to reusing repos. In my experience, large repos and or mono-repos have many benefit e.g. not duplicating CI code, atomic changes across projects, ...

In the aftermath, I would now suggest the same for `mozilla/mtu`.  E.g. in the case of `mozilla/mtu` ~10 out of the total 17 pull requests are about copying the CI environment that we already have on `mozilla/neqo` or that we would want in `mozilla/neqo`.

https://github.com/mozilla/mtu/pulls?q=is%3Apr+is%3Aclosed

That is not to say that it shouldn't be its own crate.

In this concrete case, if `mozilla/mtu` would live in `mozilla/neqo` I could add the above suggested `target` input to the Rust action and update its usage in a single PR.

I don't think we should be dogmatic on either mono-repo vs many-repos. That said, I want to make sure we give the overhead introduced through having many small repos enough weight.